frontend/src/metabase/lib/i18n.js

```diff
@@ -2,18 +2,22 @@ import dayjs from "dayjs";
 import moment from "moment-timezone"; // eslint-disable-line no-restricted-imports -- deprecated usage
 import { addLocale, useLocale } from "ttag";
 
+import api from "metabase/lib/api";
 import { DAY_OF_WEEK_OPTIONS } from "metabase/lib/date-time";
 import MetabaseSettings from "metabase/lib/settings";
 
 // note this won't refresh strings that are evaluated at load time
 export async function loadLocalization(locale) {
   // we need to be sure to set the initial localization before loading any files
   // so load metabase/services only when we need it
-  const { I18NApi } = require("metabase/services");
   // load and parse the locale
   const translationsObject =
     locale !== "en"
-      ? await I18NApi.locale({ locale })
+      ? // ? await I18NApi.locale({ locale })
+        // the function above uses GET, which adds custom headers -> we run into CORS
+        await fetch(`${api.basename}/app/locales/${locale}.json`).then(
```

Comment:
I don't think this changes anything. Adding a header or not, requesting a resource from a cross-site endpoint should trigger CORS mechanism.

I couldn't repro this problem since I couldn't even run the SDK with this branch.

Testing with this PR vs `master` @ 688b9ad95da7ae9fc87d39b816d71477edc5723, I can run the SDK with master.

Reply:
[Simple requests don't trigger preflight](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests), even though they still need the header to access the response. Using our GET helper we're adding headers and other custom stuff -> not a simple request anymore.
I've tested it locally again and it won't work without this change, it will trigger a pre-flight and our BE doesn't support pre-flights on static assets, but it will return the CORS header just fine on the normal request 🤷
I'm not sure exactly what should be the "correct" way to handle it, I'm not sure if it's normal to have pre-flight for static assets, but skipping pre-flights should make the loading a tiny bit faster anyway

Reply:
So you can run the SDK with this PR? My issue was that I couldn't run this (well the SDK is compiled, but it seems there are some BE errors that cause the endpoint to not work with the SDK), so I couldn't really test anything.

Reply:
I rebases it on master and yes I can run it (I think even before the rebase), what error did you see?

Reply:
BE
```
2024-10-01 11:50:47,244 DEBUG middleware.log :: GET /api/session/properties 200 33.4 ms (1 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (2 idle, 0 queued) (62 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
2024-10-01 11:50:47,261 DEBUG middleware.log :: GET /api/user/current 401 570.0 µs (0 DB calls) {:metabase-user-id nil}
"Unauthenticated"

2024-10-01 11:50:53,220 INFO middleware.exceptions :: Request canceled before finishing.
2024-10-01 11:50:53,234 DEBUG middleware.log :: OPTIONS /api/session/properties 200 934.0 µs (0 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (4 idle, 0 queued) (70 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
2024-10-01 11:50:53,238 INFO middleware.exceptions :: Request canceled before finishing.
2024-10-01 11:50:53,247 INFO middleware.exceptions :: Request canceled before finishing.
2024-10-01 11:50:53,267 DEBUG middleware.log :: OPTIONS /api/user/current 200 185.3 µs (0 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (4 idle, 0 queued) (71 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
2024-10-01 11:50:58,082 DEBUG middleware.log :: OPTIONS /api/session/properties 200 195.9 µs (0 DB calls) App DB connections: 0/7 Jetty threads: 6/50 (4 idle, 0 queued) (71 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
2024-10-01 11:50:58,121 INFO middleware.exceptions :: Request canceled before finishing.
2024-10-01 11:50:58,142 DEBUG middleware.log :: OPTIONS /api/user/current 200 178.5 µs (0 DB calls) App DB connections: 0/7 Jetty threads: 5/50 (4 idle, 0 queued) (71 total active threads) Queries in flight: 0 (0 queued) {:metabase-user-id nil}
```

I believe it's BE failure, but I haven't dug deeper, basically. That caused CORS issue on FE (if BE failed ofc)
![image](https://github.com/user-attachments/assets/82e1a70d-b363-4fc6-9794-0c39361c88a7)

Reply:
Mmm I think those logs are just caused by cors not working 🤔

Reply:
@WiNloSt I tested the PR locally and it works fine with Shoppy, using the German locale. Looks like something is off in your CORS configuration, are you using localhost, enterprise edition and the right branch for MB?

![CleanShot 2567-10-03 at 06 25 09@2x](https://github.com/user-attachments/assets/a95a0363-0689-4f2c-9254-1935e36bba24)